src/proposal.ts

```diff
@@ -269,32 +259,47 @@ export class Proposal implements IStateful<IProposalState> {
         stakesFor: Number(item.stakesFor),
         title: item.title,
         url: item.url,
-        votesAgainst: item.votesFor,
-        votesFor: item.votesAgainst,
+        votesAgainst: Number(item.votesAgainst),
+        votesFor: Number(item.votesFor),
         winningOutcome: item.winningOutcome
       }
     }
 
+    // TODO: the 'no-cache' statement here is suspicious, check if it is really needed and why
     this.state = context._getObservableObject(query, itemMap, { fetchPolicy: 'no-cache' }) as Observable<IProposalState>
   }
 
-  public votes(options: IVoteQueryOptions = {}): Observable<IVote[]> {
-    options.proposal = this.id
-    return Vote.search(this.context, options)
-  }
-
-  public vote(outcome: ProposalOutcome, amount: number = 0): Operation<Vote> {
-
+  /**
+   * [votingMachine description]
+   * @return [description]
+   */
+  public votingMachine() {
+    // TODO: get the contract not from migration.json but from the proposal itself!
     // TODO: cf next two lines from alchemy on how to get the votingMacchineAddress
     // (does not work with new contract versions anymore, though, it seems)
     // const contributionRewardInstance = this.dao.getContract('ContributionReward')
     // const result = await contributionRewardInstance.methods.parameters(this.dao.address).call()
     // const votingMachineAddress = (
     //   await contributionRewardInstance.methods.getSchemeParameters(daoAvatarAddress)).votingMachineAddress
-
     // the graph indexes it at contributionRewardProposal.votingMachine, but not on the proposal entity
-    // const votingMachine = this.dao.getContract('AbsoluteVote')
-    const votingMachine = this.context.getContract('GenesisProtocol')
+    return this.context.getContract('GenesisProtocol')
```

Comment:
hmm, with this code we will only support the latest deployed GenesisProtocol, just good to be aware of this

Reply:
yes, indeed